src/main.ts

```diff
@@ -657,12 +657,22 @@ export async function getToken(input: CspInput): Promise<string> {
 export async function loadAllData(executionGraph: Object): Promise<string[]> {
   let files: string[] = []
 
+  const onlyUploadOnFailure = core.getInput("only-upload-on-failed-tasks")
+
   //TODO assertions
   for (const task of executionGraph["tasks"]) {
     if (task["status"] === "SKIPPED") {
       continue
     }
 
+    if (task["status"] === "PASSED" && onlyUploadOnFailure === "true") {
```

Comment:
This check is wrong. Instead of testing the task status field, you need to check the passed field, like we are doing when [counting how many actions passed](https://github.com/vmware-labs/vmware-image-builder-action/blob/main/src/main.ts#L381).